internal/app/admin/http/handler.go

```diff
@@ -102,45 +102,62 @@ func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc
 	}
 }
 
-// TODO(lisalu): Support dump of matching resources when cache key regex is provided.
+func printCacheEntry(key string, cache cache.ReadOnlyCache, w http.ResponseWriter) {
```

Comment:
i would suggest changing the signature to `key string[]` and condense the multiple return paths in `cacheDumpHandler`